library/std/src/sys/pal/wasi/fs.rs

```diff
@@ -563,18 +616,18 @@ pub fn readdir(p: &Path) -> io::Result<ReadDir> {
     let mut opts = OpenOptions::new();
     opts.directory(true);
     opts.read(true);
-    let dir = File::open(p, &opts)?;
+    let dir = run_path_with_cstr(p, &|p| File::open_native(p, &opts))?;
     Ok(ReadDir::new(dir, p.to_path_buf()))
 }
 
-pub fn unlink(p: &Path) -> io::Result<()> {
+pub fn unlink(p: &CStr) -> io::Result<()> {
     let (dir, file) = open_parent(p)?;
     dir.unlink_file(osstr2str(file.as_ref())?)
 }
 
-pub fn rename(old: &Path, new: &Path) -> io::Result<()> {
```

Comment:
Why is only one changed to `CStr` but not both? In fact shouldn't all `&Path` in this file be changed?